### PR TITLE
Dmp 657.add document

### PR DESCRIPTION
**Pull request by cmagowan-hmcts**
https://tools.hmcts.net/jira/browse/DMP-657

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/dailylist/DailyListRoute.java [hunk 1]
```diff
@@ -1,19 +1,47 @@
...
 
+    private final DartsFeignClient dartsFeignClient;
     private final XmlParser xmlParser;
+    private final DailyListRequestMapper dailyListRequestMapper;
+
+    public AddDocumentResponse handle(String document, String type) {
+        Optional<SystemType> systemType = SystemType.getByType(type);
+        if (systemType.isEmpty()) {
+            throw new DartsValidationException("SystemType is not valid " + type);
+        }
```
Comment (davet1985) [inline src/main/java/uk/gov/hmcts/darts/dailylist/DailyListRoute.java:29]: What's `systemType` here?

Comment (davet1985) [reply src/main/java/uk/gov/hmcts/darts/dailylist/DailyListRoute.java:29]: Forget that, worked it out now 🤦

File: src/main/resources/openapi/dailyList.yaml [hunk 2]
```diff
@@ -75,7 +79,28 @@ paths:
 
         '500':
           description: Internal Server Error
+          content:
+            application/json+problem:
+              schema:
+                $ref: './problem.yaml'
 
+  /dailylists/housekeeping:
```
Comment (davet1985) [inline src/main/resources/openapi/dailyList.yaml:87]: Not sure we need the housekeeping part in gateway?

File: src/main/java/uk/gov/hmcts/darts/dailylist/DailyListRoute.java [hunk 3]
```diff
@@ -1,19 +1,47 @@
...
+        DailyListStructure legacyDailyList = xmlParser.unmarshal(document, DailyListStructure.class);
+        DailyList modernisedDailyList = dailyListRequestMapper.mapToEntity(legacyDailyList);
+        dartsFeignClient.postDailyLists(
```
Comment (davet1985) [inline src/main/java/uk/gov/hmcts/darts/dailylist/DailyListRoute.java:32]: I might be missing it, but is there a part that validates the XML document, if it's enabled?

File: src/main/resources/openapi/dailyList.yaml (hunk 2 above)
Comment (cmagowan-hmcts) [reply src/main/resources/openapi/dailyList.yaml:87]: this is just the dailyList.yaml from darts-api, it's not generating the endpoints here, but is used for generating the request objects for gateway calling darts. i don't think we should be changing this yaml from what is in darts-api. its like having a xsd for the system we are trying to call.

File: src/main/java/uk/gov/hmcts/darts/dailylist/DailyListRoute.java (hunk 3 above)
Comment (cmagowan-hmcts) [reply src/main/java/uk/gov/hmcts/darts/dailylist/DailyListRoute.java:32]: the xmlParser on line 30 wont be able to unmarshal the XML if its invalid.

Comment (davet1985) [reply src/main/java/uk/gov/hmcts/darts/dailylist/DailyListRoute.java:32]: I mean validation against the XSD, not malformed

File: src/main/resources/openapi/dailyList.yaml (hunk 2 above)
Comment (davet1985) [reply src/main/resources/openapi/dailyList.yaml:87]: Cool, makes sense.

File: src/main/java/generated/uk/gov/govtalk/people/bs7666/BLPUextentStructure.java [hunk 4]
```diff
@@ -0,0 +1,260 @@
+//
+// This file was generated by the Eclipse Implementation of JAXB, v4.0.3
+// See https://eclipse-ee4j.github.io/jaxb-ri
+// Any modifications to this file will be lost upon recompilation of the source schema.
+//
+
+
+package uk.gov.govtalk.people.bs7666;
```
Comment (HMCTSDeveloper) [inline src/main/java/generated/uk/gov/govtalk/people/bs7666/BLPUextentStructure.java:8]: is this package name correct ? seems weird to have numbers.

**Review by HMCTSDeveloper — APPROVED** [review]

File: src/main/java/generated/uk/gov/govtalk/people/bs7666/BLPUextentStructure.java (hunk 4 above)
Comment (cmagowan-hmcts) [reply src/main/java/generated/uk/gov/govtalk/people/bs7666/BLPUextentStructure.java:8]: its generated, so guess it must be

**Pull request merged**